app/gui2/src/components/widgets/NumericInputWidget.vue

```diff
@@ -1,10 +1,24 @@
 <script setup lang="ts">
 import { PointerButtonMask, usePointer, useResizeObserver } from '@/composables/events'
+import { blurIfNecessary } from '@/util/autoBlur'
 import { getTextWidth } from '@/util/measurement'
-import { computed, ref, type StyleValue } from 'vue'
-
-const props = defineProps<{ modelValue: number; min: number; max: number }>()
-const emit = defineEmits<{ 'update:modelValue': [modelValue: number] }>()
+import { computed, ref, watch, type StyleValue } from 'vue'
+
+const props = defineProps<{
+  modelValue: number | string
+  limits?: { min: number; max: number } | undefined
+}>()
+const emit = defineEmits<{ 'update:modelValue': [modelValue: number | string] }>()
+
+const inputFieldActive = ref(false)
+const currentInputValue = ref(props.modelValue)
+watch(
+  () => props.modelValue,
+  (newValue) => {
+    currentInputValue.value = newValue
+  },
+)
+const SLIDER_INPUT_THRESHOLD = 4.0
```

Comment:
It looks a bit like antipattern of making ref reflecting `modelValue`. I would rename `currentInputValue` to something like `editedValue`... or just add a comment why we need it.